### PR TITLE
Fixing op id generation race condition

### PR DESCRIPTION
**Pull request by mo-tenstorrent**
### Ticket
#12089 

### Problem description
increment and then get on an atomic is not atomic

### What's changed
Use fetch_add to guarantee the entire process is atomic 

### Checklist
- [x] Post commit CI passes https://github.com/tenstorrent/tt-metal/actions/runs/10637840365
- [x] T3K profiler https://github.com/tenstorrent/tt-metal/actions/runs/10637838787
- [x] Device perf https://github.com/tenstorrent/tt-metal/actions/runs/10637834190
- [x] Models run https://github.com/tenstorrent/tt-metal/actions/runs/10689872490 ( Failing on what main is failing on)

### CODE REVIEW

File: ttnn/tracy/__init__.py [hunk 1]
```diff
@@ -183,164 +183,3 @@ def get_available_port():
 
 def split_comma_list(option, opt, value, parser):
     setattr(parser.values, option.dest, value.split(","))
-
```
Comment (ayerofieiev-tt) [inline ttnn/tracy/__init__.py:186]: not needed anymore?

File: ttnn/cpp/ttnn/tensor/tensor.cpp [hunk 2]
```diff
@@ -805,7 +805,7 @@ Tensor set_tensor_id(const Tensor& tensor) {
         return tensor;
     }
     auto output = tensor;
-    ttnn::increment_tensor_id();
+    auto id = ttnn::fetch_and_increment_tensor_id();
     output.tensor_id = ttnn::get_tensor_id();
```
Comment (dmakoviichuk-tt) [inline ttnn/cpp/ttnn/tensor/tensor.cpp:809]: should it be output.tensor_id = ttnn::fetch_and_increment_tensor_id();?

File: ttnn/cpp/ttnn/core.cpp [hunk 3]
```diff
@@ -7,18 +7,18 @@
 
 namespace ttnn {
 
-std::atomic<std::int64_t> TENSOR_ID = -1;
-std::atomic<std::int64_t> PYTHON_OPERATION_ID = -1;
-std::atomic<std::int64_t> DEVICE_OPERATION_ID = -1;
+std::atomic<std::int64_t> TENSOR_ID = 0;
```
Comment (dmakoviichuk-tt) [inline ttnn/cpp/ttnn/core.cpp:10]: Nice fix!
could you create singleton and put all global variables inside so there is no direct access from the outside?
I saw this code and that was my plan.

File: ttnn/tracy/__init__.py (hunk 1 above)
Comment (mo-tenstorrent) [reply ttnn/tracy/__init__.py:186]: This was all defined in `__main__.py` it was all repeated here.

File: ttnn/cpp/ttnn/core.cpp (hunk 3 above)
Comment (mo-tenstorrent) [reply ttnn/cpp/ttnn/core.cpp:10]: Sure, does calling the singleton `CoreIDs` make sense?

Comment (dmakoviichuk-tt) [reply ttnn/cpp/ttnn/core.cpp:10]: CoreIDs or GlobalCoreIDs sounds good for me.

**Review by dmakoviichuk-tt — APPROVED** [review]

**Review by eyonland — APPROVED** [review]

**Review by cfjchu — APPROVED** [review]

**Pull request merged**